src/algebra/module/submodule/basic.lean

```diff
@@ -257,6 +258,34 @@ lemma injective_subtype : injective p.subtype := subtype.coe_injective
 @[simp] lemma coe_sum (x : ι → p) (s : finset ι) : ↑(∑ i in s, x i) = ∑ i in s, (x i : M) :=
 map_sum p.subtype _ _
 
+/-! ### Actions by `submodule`s
```

Comment:
Would you mind splitting this to a standalone PR?

Reply:
I have made two PRs #18814 and #18815 with the results on submodule and basis with restricted scalars respectively.  

BTW, you are probably aware of that but there seems to be something wrong with CI at the moment, it always fail during the lint style check while installing `bibtool`, see https://github.com/leanprover-community/mathlib/actions/runs/4708837338/jobs/8351530017#step:3:66